api/app/reviews/tests.py

```diff
@@ -1465,27 +1439,47 @@ def static_seed_data(self):
         db.session.commit()
 
         self.first_headers = self.get_auth_header_for("firstuser@mail.com")
-        # self.other_headers = self.get_auth_header_for("someuser@mail.com")
 
         db.session.flush()
 
     def test_get_reference_request_by_response_id(self):
+        self.seed_static_data()
+        reference_req = ReferenceRequest(1, 'Mr', 'John', 'Snow', 'Supervisor', 'common@email.com')
+        reference_request_repository.create(reference_req)
+        REFERENCE_DETAIL = {
+            'token': reference_req.token,
+            'uploaded_document': 'DOCT-UPLOAD-78999',
+        }
+        # response = self.app.get(
+        #     '/api/v1/reference', data=REFERENCE_DETAIL, headers=self.first_headers)
+        # self.assertEqual(response.status_code, 201)
+
+        params = {'event_id': 1, 'application_form_id': 1, 'user_id:': 5, 'sort_column' : 'review_response_id',
+                  'reference_detail': REFERENCE_DETAIL} # confirm params, check ids
+
+        response = self.app.get('/api/v1/review', headers=self.get_auth_header_for('r1@r.com'), data=params)

```

Comment:
I am still struggling with this line. 

If I understand correctly, this is where we return the `data` as described in your previous comment. This comes from the different models - Review and Reference and will be found using the "response_id" as the foreign keys?

Reply:
That's correct, but remember it's the API's job to do that joining between Review and Reference, not the test... the test is to ensure that the API does it correctly, perhaps its confusing that the API doesn't currently do it, but when you're writing the test you need to pretend that you've finished the task and are now checking that it was done correctly :)

Reply:
Ah, yes. Okay. It's clicked for me now :D Thank you!